common/extension/config_load_processor.go

```diff
@@ -0,0 +1,135 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package extension
+
+import (
+	"github.com/apache/dubbo-go/common"
+	"github.com/apache/dubbo-go/common/constant"
+	"github.com/apache/dubbo-go/config/interfaces"
+	"reflect"
+)
+
+const (
+	LoadProcessReferenceConfigFunctionName        = "LoadProcessReferenceConfig"
+	LoadProcessServiceConfigFunctionName          = "LoadProcessServiceConfig"
+	AfterAllReferencesConnectCompleteFunctionName = "AfterAllReferencesConnectComplete"
+	AfterAllServicesListenCompleteFunctionName    = "AfterAllServicesListenComplete"
+	BeforeShutdownFunctionName                    = "BeforeShutdown"
+)
+
+var (
+	loadProcessors      = make(map[string]interfaces.ConfigLoadProcessor)
+	loadProcessorValues = make(map[string]reflect.Value)
+
+	referenceURL = make(map[string][]*common.URL)
+	serviceURL   = make(map[string][]*common.URL)
```

Comment:
Should you add a read/write lock for this variables??

Reply:
Done